src/middlewared/middlewared/plugins/pool_/snapshot_count.py

```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+
+from middlewared.schema import accepts, returns, Int, Str
+from middlewared.service import Service
+
+
+class PoolDatasetService(Service):
+
+    class Config:
+        namespace = "pool.dataset"
+
+    @accepts(Str("dataset"))
+    @returns(Int())
+    def snapshot_count(self, dataset):
+        mountpoint = Path("/mnt") / dataset
```

Comment:
We might be able to look up through getmntent in middlewared/utils. c.f. zfs.py

Reply:
The .zfs/snapshot directory has a hardcoded inode number. We can double-check that as well.